infrastructure/smart-contracts/contracts/mock/MockRollup.sol

```diff
@@ -0,0 +1,143 @@
+
+contract MockRollup {
+    // Variables to track the latest node created and the latest confirmed node
+    uint64 public latestNodeCreated; // Tracks the most recently created node number
+    uint64 public latestConfirmed;   // Tracks the most recently confirmed node number
+
+    struct Node {
+        bytes32 stateHash;                 // Hash representing the state of the chain at this node
+        bytes32 challengeHash;             // Hash of data that can be challenged
+        bytes32 confirmData;               // Hash of data that will be confirmed when this node is confirmed
+        uint64 prevNum;                    // Index of the node preceding this one
+        uint64 deadlineBlock;              // Block number after which this node can be confirmed
+        uint64 noChildConfirmedBeforeBlock;// Block number before which a child node cannot be confirmed
+        uint64 stakerCount;                // Number of stakers (including zombies) on this node
+        uint64 childStakerCount;           // Number of stakers (including zombies) on a child node
+        uint64 firstChildBlock;            // Block number of the first child of this node
+        uint64 latestChildNumber;          // Number of the most recent child created from this node
+        uint64 createdAtBlock;             // Block number when this node was created
+        bytes32 nodeHash;                  // Hash summarizing all critical data to ensure node validity, protect against reorgs
+    }
+
+    // Mapping from node number to Node struct to store all created nodes
+    mapping(uint64 => Node) private _nodes;
+
+    struct GlobalState {
+        bytes32 bytes32Vals;               // Example field for global state; adjust as needed for real use cases
+        uint64[2] u64Vals;                 // Example array for storing two 64-bit unsigned integer values
+    }
+
+    struct Assertion {
+        GlobalState beforeState;           // Represents the global state before a transition
+        GlobalState afterState;            // Represents the global state after a transition
+        uint64 numBlocks;                  // Number of blocks involved in this assertion
+    }
+
+    // Events to log node creation and confirmation actions
+    event NodeCreated(
+        uint64 indexed nodeNum,
+        bytes32 indexed parentNodeHash,
+        bytes32 indexed nodeHash,
+        bytes32 executionHash,
+        Assertion assertion,
+        bytes32 afterInboxBatchAcc,
+        bytes32 wasmModuleRoot,
+        uint256 inboxMaxCount
+    );
+
+    event NodeConfirmed(uint64 indexed nodeNum, bytes32 blockHash, bytes32 sendRoot);
+
+    /**
+     * @notice Creates a new node with provided block hash and send root.
+     * @param blockHash The simulated block hash to associate with this node.
+     * @param sendRoot The simulated send root to associate with this node.
+     */
+    function createNode(
+        bytes32 blockHash,
+        bytes32 sendRoot
+    ) external {
```

Comment:
This will need to have the assertionId as a param so we can set it and mock mainnet NodeCreated events

Reply:
This has been updated and re-deployed on [Sepolia here.](https://sepolia.arbiscan.io/address/0xd947360f87B9f01Eacd02041FfeE91D16065e64b)

Also deployed a new subgraph, that is [currently syncing here](https://subgraph.satsuma-prod.com/xai/sentry-sepolia/version/0.0.25-sepolia-mock-102/playground).